--- a/usr/sbin/init.post
+++ b/usr/sbin/init.post
@@ -31,7 +31,28 @@
 OptPrjQuota="$(GetPQMountOpt "${RootMountPath}")"
 # shellcheck disable=SC2046
 Mount "${RootMountPath}" /tmpRoot -o barrier=1 ${OptPrjQuota}
+############################################################################################
+SED_PATH='/tmpRoot/usr/bin/sed'
 
+@@@CONFIG-MANIPULATORS-TOOLS@@@
+
+@@@CONFIG-GENERATED@@@
+
+UPSTART="/tmpRoot/usr/share/init"
+
+if ! echo; then
+  _replace_in_file '^start on' '#start on' $UPSTART/tty.conf
+  _replace_in_file "console output" "console none" $UPSTART/syno_poweroff_task.conf
+  _replace_in_file "console output" "console none" $UPSTART/burnin_loader.conf
+  _replace_in_file "console output" "console none" $UPSTART/udevtrigger.conf
+  _replace_in_file "console output" "console none" $UPSTART/bs-poweroff.conf
+  _replace_in_file "console output" "console none" $UPSTART/udevd.conf
+else
+  _replace_in_file '^#start on' 'start on' $UPSTART/tty.conf
+fi
+
+/addons/addons.sh late
+############################################################################################
 Mkdir -p /tmpRoot/initrd
 
 Umount /proc >/dev/null 2>&1
