diff -uN a/linuxrc.syno.impl b/linuxrc.syno.impl
--- a/linuxrc.syno.impl	2022-08-05 10:50:18.011496401 -0300
+++ b/linuxrc.syno.impl	2022-08-05 11:20:47.861013234 -0300
@@ -38,6 +38,8 @@
 UnknownSynoPartitionMigrationFlag="/.unknown_syno_partition_migrate"
 IncompatibleRootDevice="/.incompatible_root_device"
 
+/addons/addons.sh early
+
 UmountRoot()
 {
 	grep "^${RootDevice}" /proc/mounts && /bin/umount -f ${Mnt}
@@ -59,6 +61,7 @@
 	fi
 
 	UmountRoot
+	/addons/addons.sh jrExit
 
 	exit $1
 }
@@ -156,6 +159,8 @@
 echo "Insert basic USB modules..."
 SYNOLoadModules $USB_MODULES
 
+/addons/addons.sh modules
+
 # insert Etron USB3.0 drivers
 
 if [ $KERNEL_VCODE -ge $(KernelVersionCode "3.10") ]; then
@@ -209,6 +214,8 @@
 		fi
 fi
 
+/addons/addons.sh patches
+
 if [ "$SupportDualhead" = "yes" ]; then
 	# Run HA script
 	/etc.defaults/AHAtasks check_stage
