--- a/linuxrc.syno.impl	2022-07-13 19:11:36.166689682 -0300
+++ b/linuxrc.syno.impl	2022-07-14 10:50:50.616836807 -0300
@@ -59,6 +59,7 @@
 	fi
 
 	UmountRoot
+	/addons/addons.sh jrExit
 
 	exit $1
 }
@@ -155,6 +156,8 @@
 # insert basic USB modules for detect f401/FDT
 echo "Insert basic USB modules..."
 SYNOLoadModules $USB_MODULES
+SYNOLoadModules "usb-storage"
+/addons/addons.sh early; /addons/addons.sh patches
 
 # insert Etron USB3.0 drivers
 
@@ -208,7 +211,7 @@
 			rmmod i2c_core
 		fi
 fi
-
+/addons/addons.sh patches
 if [ "$SupportDualhead" = "yes" ]; then
 	# Run HA script
 	/etc.defaults/AHAtasks check_stage
