--- a/linuxrc.syno.impl
+++ b/linuxrc.syno.impl
@@ -38,6 +38,8 @@
 UnknownSynoPartitionMigrationFlag="/.unknown_syno_partition_migrate"
 IncompatibleRootDevice="/.incompatible_root_device"
 
+/addons/addons.sh early
+
 UmountRoot()
 {
 	grep "^${RootDevice}" /proc/mounts && /bin/umount -f ${Mnt}
@@ -59,5 +61,6 @@
 	fi
 
+	/addons/addons.sh jrExit
 	UmountRoot
 
 	exit $1
 }
@@ -155,6 +158,8 @@
 # insert basic USB modules for detect f401/FDT
 echo "Insert basic USB modules..."
 SYNOLoadModules $USB_MODULES
+
+/addons/addons.sh modules

 # insert Etron USB3.0 drivers
 
@@ -209,6 +214,8 @@
 		fi
 fi
 
+/addons/addons.sh patches
+
 if [ "$SupportDualhead" = "yes" ]; then
 	# Run HA script
 	/etc.defaults/AHAtasks check_stage
